extensions/resource-deployment/src/ui/deployAzureSQLVMWizard/pages/vmSettingsPage.ts

```diff
@@ -0,0 +1,410 @@
+						},
+						{
+							component: this.wizard.createFormRowComponent(view, constants.VmVersionDropdownLabel, '', this._vmImageVersionDropdownLoader, true)
+						},
+						{
+							component: this.wizard.createFormRowComponent(view, constants.VmSizeDropdownLabel, '', this._vmSizeDropdownLoader, true)
+						}
+					],
+					{
+						horizontal: false,
+						componentWidth: '100%'
+					})
+				.withLayout({ width: '100%' })
+				.component();
+
+
+			return view.initializeModel(this._form);
+		});
+	}
+
+	public async onEnter(): Promise<void> {
+		this.populateVmImageDropdown();
+		this.populateVmSizeDropdown();
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+
+			if (pcInfo.newPage < pcInfo.lastPage) {
+				return true;
+			}
+
+			let showErrorMessage = '';
+			/**
+			 * VM name rules:
+			 * 	1. 1-15 characters
+			 *  2. Cannot contain only numbers
+			 *  3. Cannot start with underscore and end with period or hyphen
+			 *  4. Virtual machine name cannot contain special characters \/""[]:|<>+=;,?*
+			 */
+			let vmname = this.wizard.model.vmName;
+			if (vmname.length < 1 && vmname.length > 15) {
+				showErrorMessage += 'Virtual machine name must be between 1 and 15 characters long.\n';
+			}
+			if (/^\d+$/.test(vmname)) {
+				showErrorMessage += 'Virtual machine name cannot contain only numbers.\n';
+			}
+			if (vmname.charAt(0) === '_' || vmname.slice(-1) === '.' || vmname.slice(-1) === '-') {
+				showErrorMessage += 'Virtual machine name Can\'t start with underscore. Can\'t end with period or hyphen\n';
+			}
+			if (/[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/.test(vmname)) {
+				showErrorMessage += 'Virtual machine name cannot contain special characters \/""[]:|<>+=;,?* .\n';
+			}
+
+
+			/**
+			 * VM admin/root username rules:
+			 *  1. 1-20 characters long
+			 *  2. cannot contain special characters \/""[]:|<>+=;,?*
+			 */
+			let username = this.wizard.model.vmUsername;
+			if (username.length < 1 || username.length > 20) {
+				showErrorMessage += 'Username must be between 1 and 20 characters long.\n';
+			}
+			if (/[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?]+/.test(username)) {
+				showErrorMessage += 'Username cannot contain special characters \/""[]:|<>+=;,?* .\n';
+			}
+
+			showErrorMessage += this.wizard.validatePassword(this.wizard.model.vmPassword);
+
+			if (this.wizard.model.vmPassword !== this._adminComfirmPasswordTextBox.value) {
+				showErrorMessage += 'Password and confirm password must match.\n';
+			}
+
+			if (this._vmSize.includes((this._vmSizeDropdown.value as azdata.CategoryValue).name)) {
+				showErrorMessage += 'Select a virtual machine size.\n';
+			}
+
+			this.wizard.showErrorMessage(showErrorMessage);
+
+			if (showErrorMessage !== '') {
+				return false;
+			}
+			return true;
+		});
+	}
+
+	public onLeave(): void {
+		this.wizard.wizardObject.registerNavigationValidator((pcInfo) => {
+			return false;
```

Comment:
should this be true here? if next page doesn't register a validator, you won't be able to move next.